docs/proposal/1/design.md

```diff
@@ -61,61 +61,63 @@ A lightweight **ServiceContainer** will provide access to cross-cutting concerns
 
 ### 7. Data Persistence & State Transfer
 
-To avoid the pitfalls of shared mutable global state and ensure data integrity, we will adopt a **Hydration/Dehydration** model for persistence.
+To avoid the pitfalls of shared mutable global state while solving the performance and boilerplate issues of strict DTOs, we will adopt a **Manifest & Protocol** model.
 
 #### 7.1. Core Principles
 
-- **No Global Session in ECS**: Systems inside a Scene **never** access a global `GameSession`, `SaveFile`, or `Database` object directly. They operate exclusively on local Components.
-- **Explicit Data Transfer**: Data moving between the permanent storage (Session) and the active game (Scene) is always explicit, typed, and snapshotted.
-- **Source of Truth**:
-    - During a Scene's lifecycle, the **ECS World is the single source of truth**.
-    - The persistent storage is updated only during transition boundaries or explicit checkpoints, preventing desync bugs.
+- **No Global Session in ECS**: Systems inside a Scene **never** access a global `GameSession` directly.
+- **Serializable Protocol**: Instead of separate DTO classes, Components that require persistence must implement a `Serializable` protocol (e.g., `to_dict()` and `from_dict()`). This eliminates duplicate class definitions.
```

Comment:
Instead of an `Serializable` protocol (e.g., `to_dict()` and `from_dict()`), could we should use the MsgSpec library that we already pull in and use?